examples/08_join_aggregation.py

```diff
@@ -1,302 +1,548 @@
+        (TargetEncoder(), ["goods_code"]),
+        ("passthrough", ["ID"]),
+    ],
+)
 
-    movie_avg_rating = X_train.groupby("movieId")["y"].mean().to_frame().reset_index()
+products_transformed = vectorizer.fit_transform(*get_X_y(products))
+time_to_fit = time() - tic
 
-    y_pred = X_test.merge(movie_avg_rating, on="movieId", how="left")["y"]
-    y_pred = y_pred.fillna(y_pred.mean())
+TableReport(products_transformed)
 
-    return r2_score(y_true=y_test, y_pred=y_pred)
+# %%
+# Let's now detail how to leverage |AggJoiner| here. We have just encoded each product
+# attributes, and now we need to somehow aggregate these product encodings into their
+# respective baskets.
+#
+# By aggregating instead of concatenating, we obtain an invariant number of columns,
+# and we remove the sparsity of the dataset.
+#
+# But which aggregation operation should we choose? Since the |MinHashEncoder| hashes
+# ngrams with different hashing functions and return their minimum, it makes sense to
+# aggregate different product encodings using their **minimum** for each dimension.
+# You can view MinHash minimums as activations.
+#
+# For numeric columns and columns encoded with the |TargetEncoder|, we take the mean,
+# standard deviation, minimum and maximum to extract a representative summary of each
+# distribution.
+#
+# We can apply these two sets of operations by chaining together two |AggJoiner| in
+# a |Pipeline| using |make_pipeline|. We also make use of skrub selectors to select
+# columns with the ``glob`` syntax.
+#
+# We need to pass the product dataframe as an auxiliary table argument to AggJoiner
+# in ``__init__``. The basket dataframe is our main table, and we pass it during
+# ``fit``. We discuss the limitations of this design in the conclusion at the bottom
+# of this notebook.
+#
+# Let's display the output of this preprocessing pipeline.
 
+from sklearn.pipeline import make_pipeline
 
-all_baseline_r2 = []
-for train_idx, test_idx in TimeSeriesSplit(n_splits=10).split(X, y):
-    all_baseline_r2.append(baseline_r2(X, y, train_idx, test_idx))
+from skrub import AggJoiner
+from skrub import _selectors as s
+
+minhash_cols = "ID" | s.glob("item_*") | s.glob("model_*") | s.glob("make_*")
+single_cols = ["ID", "goods_code", "Nbr_of_prod_purchas", "cash_price"]
+
+pipe_agg_joiner = make_pipeline(
+    AggJoiner(
+        aux_table=s.select(products_transformed, minhash_cols),
+        key="ID",
+        operations=["min"],
+    ),
+    AggJoiner(
+        aux_table=s.select(products_transformed, single_cols),
+        key="ID",
+        operations=["mean", "sum", "std", "min", "max"],
+    ),
+)
+basket_train_transformed = pipe_agg_joiner.fit_transform(baskets_train)
+
+TableReport(basket_train_transformed)
+
+# %%
+# Now that we get a sense of how the |AggJoiner| can help us, we complete this pipeline
+# with a |HGBC| and evaluate our final model.
+
+tic = time()
+agg_join_estimator = make_pipeline(
+    pipe_agg_joiner,
```

Comment:
maybe we could append to the pipeline instead of wrapping in a new one? or we can keep it like this now and it will just become a `recipe.add()` later (where later is before the next release anyway)

Reply:
Yes, demos like this are a good candidate for the recipe